skimage/exposure/_adapthist.py

```diff
@@ -147,13 +158,11 @@ def _clahe(image, kernel_size, clip_limit, nbins):
 
     # calculate graylevel mappings for each contextual region
     # rearrange image into flattened contextual regions
-    ns_hist = [int(s / k) - 1 for s, k in zip(image.shape, kernel_size)]
+    ns_hist = [int(s / k) for s, k in zip(image.shape, kernel_size)]
```

Comment:
why does the `-1` go away here?

Reply:
In the new scheme, the histogram needs to be calculated for an additional block along each dimension. See the illustration in my comment below.